packages/web3-providers-ws/src/index.js

```diff
@@ -392,8 +392,12 @@ WebsocketProvider.prototype.reconnect = function () {
 
     if (this.responseQueue.size > 0) {
         this.responseQueue.forEach(function (request, key) {
-            request.callback(errors.PendingRequestsOnReconnectingError());
-            _this.responseQueue.delete(key);
+            try {
+                request.callback(errors.PendingRequestsOnReconnectingError());
+            } catch (e) {
+                _this.responseQueue.delete(key);
```

Comment:
Once request is sent to eth node , its moved from requestQueue to responseQueue, and with new message/response received its id is matched with responseQueue and result is sent back to user by web3 lib.

This change will not clear `responseQueue` on reconnects in normal flow(if there is no error thrown by callback), so for any requests sent to eth node web3.js will continue keeping requests in queue.

Reply:
so with-out this change, if get the error "provider started to reconnect ...." then i should not get the receipt of pending txs right?  as "_this.responseQueue.delete(key);" will clear response queue. but after reconnect i am getting the receipt of tx. i wonder how.

Reply:
any update on this ? @jdevcs

Reply:
fixed in https://github.com/web3/web3.js/pull/5820